homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -46,14 +46,17 @@ class SpeedtestSensor(CoordinatorEntity, RestoreEntity, SensorEntity):
 
     _attr_icon = ICON
 
-    def __init__(self, coordinator: SpeedTestDataCoordinator, sensor_type: str) -> None:
+    def __init__(
+        self,
+        coordinator: SpeedTestDataCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator)
-        self.type = sensor_type
+        self.entity_description = description
 
-        self._attr_name = f"{DEFAULT_NAME} {SENSOR_TYPES[sensor_type][0]}"
-        self._attr_unit_of_measurement = SENSOR_TYPES[self.type][1]
-        self._attr_unique_id = sensor_type
+        self._attr_name = f"{DEFAULT_NAME} {description.name}"
+        self._attr_unique_id = description.key
```

Comment:
```suggestion
        self._attr_name = f"{DEFAULT_NAME} {description.name}"
        self._attr_unit_of_measurement = description.unit_of_measurement
        self._attr_unique_id = description.key
```
Looks like `unit_of_measurement` is missing

Reply:
No? The description is set:

```py
self.entity_description = description
```

So, it gets it from there.

Reply:
Nice